### PR TITLE
[WIP] Revise documentation for Swift 5 stdlib additions

### PR DESCRIPTION
**Pull request by natecook1000**
This is tracking WIP to revise the docs for the additions in [SE-0221](https://github.com/apple/swift-evolution/blob/master/proposals/0221-character-properties.md).

### CODE REVIEW

File: stdlib/public/core/CharacterProperties.swift [hunk 1]
```diff
@@ -236,25 +238,38 @@ extension Character {
...
+  /// - "®" (U+00AE REGISTERED SIGN)
+  /// - "⌹" (U+2339 APL FUNCTIONAL SYMBOL QUAD DIVIDE)
+  /// - "⡆" (U+2846 BRAILLE PATTERN DOTS-237)
   public var isSymbol: Bool {
     return _firstScalar.properties.generalCategory._isSymbol
   }
 
-  /// Whether this Character represents a symbol used mathematical formulas
+  /// A Boolean value indicating whether this character represents a symbol
+  /// that naturally appearx in mathematical contexts.
```
Comment (krilnon) [inline stdlib/public/core/CharacterProperties.swift]:
```suggestion
  /// that naturally appears in mathematical contexts.
```
x -> s

File: stdlib/public/core/CharacterProperties.swift [hunk 2]
```diff
@@ -96,7 +96,9 @@ extension Character {
   /// If this Character is a whole number, return the value it represents, else
   /// nil.
   ///
-  /// Examples:
+  /// A Boolean value indicating whether this character represents a newline.
```
Comment (Dante-Broggi) [inline stdlib/public/core/CharacterProperties.swift]: New comment describing newlines, but code is for whole numbers.

File: stdlib/public/core/CharacterProperties.swift [hunk 3]
```diff
@@ -22,46 +22,47 @@ extension Character {
...
-  ///   * U+0085: NEXT LINE (NEL)
-  ///   * U+2028: LINE SEPARATOR
-  ///   * U+2029: PARAGRAPH SEPARATOR
+  /// For example, the following characters all represent newlines:
   ///
+  /// - "\n" (U+000A): LINE FEED (LF)
+  /// - U+000B: LINE TABULATION (VT)
+  /// - U+000C: FORM FEED (FF)
+  /// - "\r" (U+000D): CARRIAGE RETURN (CR)
+  /// - "\r\n" (U+000A U+000D): CR-LF
```
[the remark this answers is not in the document]
Comment (natecook1000) [reply stdlib/public/core/CharacterProperties.swift]: Wow, great catch!

File: stdlib/public/core/Integers.swift [hunk 4]
```diff
@@ -105,41 +142,41 @@ public extension AdditiveArithmetic where Self : ExpressibleByIntegerLiteral {
...
+/// A type with values that support multiplication.
 ///
 /// The `Numeric` protocol provides a suitable basis for arithmetic on
 /// scalar values, such as integers and floating-point numbers. You can write
 /// generic methods that operate on any numeric type in the standard library
 /// by using the `Numeric` protocol as a generic constraint.
 ///
-/// The following example declares a method that calculates the total of any
-/// sequence with `Numeric` elements.
+/// The following example extends `Seqeunce` with a method that returns an
```
Comment (krilnon) [inline stdlib/public/core/Integers.swift]:
```suggestion
/// The following example extends `Sequence` with a method that returns an
```

Typo on ‘Sequence’.

**Comment by natecook1000** [comment]
These changes are ready for review:

@milseman for the Character/UnicodeScalar properties
@stephentyrone for the SIMD types/protocols
and also @airspeedswift

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

File: stdlib/public/core/SIMDVectorTypes.swift.gyb [hunk 5]
```diff
@@ -67,6 +85,7 @@ public struct SIMD${n}<Scalar> : SIMD where Scalar: SIMDScalar {
   }
 
 %  for (half,indx) in [('low','i'), ('high',str(n/2)+'+i'), ('even','2*i'), ('odd','2*i+1')]:
+  /// A half-length vector covering the ${half} elements of the vector.
```
Comment (natecook1000) [inline stdlib/public/core/SIMDVectorTypes.swift.gyb]: @stephentyrone Does this phrasing ("covering the even elements of the vector") sound okay to you? Suggestions?

Comment (stephentyrone) [reply stdlib/public/core/SIMDVectorTypes.swift.gyb]: "made up of"?

**Review by milseman — APPROVED** [review]
Character and Scalar properties LGTM

File: stdlib/public/core/SIMDVectorTypes.swift.gyb (hunk 5 above)
Comment (natecook1000) [reply stdlib/public/core/SIMDVectorTypes.swift.gyb]:
```suggestion
  /// A half-length vector made up of the ${half} elements of the vector.
```

**Review by stephentyrone — APPROVED** [review]

**Comment by natecook1000** [comment]
@swift-ci Please smoke test and merge

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

**Pull request merged**